src/DIRAC/WorkloadManagementSystem/Agent/SiteDirector.py

```diff
@@ -646,13 +646,19 @@ def _getCE(self, queue):
             ceDict["OwnerGroup"] = self.voGroups
 
         if self.checkPlatform:
-            result = self.resourcesModule.getCompatiblePlatforms(self.queueDict[queue]["ParametersDict"]["Platform"])
+            platform = self.queueDict[queue]["ParametersDict"].get("Platform")
+            if not platform:
+                self.log.error("No platform set for CE %s, returning 'ANY'" % ce)
+                ceDict["Platform"] = "ANY"
+                return ce, ceDict
+            result = self.resourcesModule.getCompatiblePlatforms(platform)
             if not result["OK"]:
                 self.log.error(
                     "Issue getting compatible platforms, returning 'ANY'",
                     "%s: %s" % (self.platforms, result["Message"]),
                 )
                 ceDict["Platform"] = "ANY"
+                return ce, ceDict
             ceDict["Platform"] = result["Value"]
```

Comment:
How about this instead:
```suggestion
            if result["OK"]:
                ceDict["Platform"] = result["Value"]
            else:
                self.log.error(
                    "Issue getting compatible platforms, returning 'ANY'",
                    "%s: %s" % (self.platforms, result["Message"]),
                )
                ceDict["Platform"] = "ANY"
```
(I know it's the same but it feels more tollerant to the function being edited in future to add more logic to the end.)